tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -837,7 +1243,26 @@ HRESULT DebuggerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler,
     mdToken exceptionToken = mdTokenNil;
     mdToken callTargetReturnToken = mdTokenNil;
     ILInstr* firstInstruction;
-    hr = debuggerTokens->ModifyLocalSigAndInitialize(&rewriterWrapper, caller, &callTargetStateIndex, &exceptionIndex,
+
+    bool isAsyncMethod;
+    auto isAsyncMethodProbeHr = IsAsyncMethodProbe(module_metadata.metadata_import, caller, isAsyncMethod);
+    IfFailRet(isAsyncMethodProbeHr);
+    TypeSignature methodReturnType{};
+    if (isAsyncMethod)
+    {
+        hr = GetTaskReturnType(rewriterWrapper.GetILRewriter()->GetILList(), module_metadata, methodLocals, &methodReturnType);
```

Comment:
Maybe you can keep aside in the preprocessing phase the `FunctionInfo` (or the `TypeSignature` of the return value) of the kickoff to better facilitate the process of determining what is the return value?

Reply:
Internal ticket ID 1051